features/update/basic_async/feature.ts

```diff
@@ -0,0 +1,55 @@
+import { WorkflowUpdateFailedError } from '@temporalio/client';
+import { Feature } from '@temporalio/harness';
+import * as wf from '@temporalio/workflow';
+import * as assert from 'assert';
+
+const WORKFLOW_INITIAL_STATE = '';
+const UPDATE_ARG = 'update-arg';
+const BAD_UPDATE_ARG = 'reject-me';
+const UPDATE_RESULT = 'update-result';
+const UPDATE_RESULT_NOT_YET_RECEIVED = 'update-result-not-yet-received';
+
+const myUpdate = wf.defineUpdate<string, [string]>('myUpdate');
+
+/**
+ * A workflow with a signal and signal validator. If accepted, the signal makes
+ * a change to workflow state. The workflow does not terminate until such a
+ * change occurs.
+ */
+export async function workflow(): Promise<string> {
+  let state = WORKFLOW_INITIAL_STATE;
+  const update = (arg: string) => {
+    state = arg;
+    return UPDATE_RESULT;
+  };
+  const validate = (arg: string) => {
+    if (arg == BAD_UPDATE_ARG) {
+      throw new Error('Invalid Update argument');
```

Comment:
You don't need to do it in these feature tests, but in the TS tests, confirm that if they use an application error it is serialized properly as the cause of the update failed error client side (allows them to add details)

Reply:
Thanks yes my tests in the main PR do this.